nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/FlowResource.java

```diff
@@ -420,7 +422,10 @@ public Response getFlowMetrics(
                     value = "The producer for flow file metrics. Each producer may have its own output format.",
                     required = true
             )
-            @PathParam("producer") final String producer) throws InterruptedException {
+            @PathParam("producer") final String producer,
+            @QueryParam("sampleName") final String sampleName,
+            @QueryParam("sampleLabelValues") final String sampleLabelValues,
+            @QueryParam("firstFieldName") @DefaultValue("samples") final String firstFieldName) throws InterruptedException {
```

Comment:
These new parameters are helpful, but they appear to be applied only to the JSON format.  Instead, they should apply to all output formats, which might require a bit more refactoring to reduce potential duplication.